### DISCUSSION TITLE
Get user_id from username

### DISCUSSION THREAD

**Discussion by alvaroserrrano** [kevinzg/facebook-scraper#371] [Q&A] [answered]
Hi, Thanks for creating this. How can we get the id of a user entering its username or even profile_name?

**Comment by neon-ninja** [comment, marked as answer]
The `get_profile` function should do it. Here's an example:

```python
from facebook_scraper import get_profile
from pprint import pprint
pprint(get_profile("zuck"))
```
outputs
```python
{'About': "I'm trying to make the world a more open place.",
 'Education': 'Harvard University\n'
              'Computer Science and Psychology\n'
              '30 August 2002 - 30 April 2004\n'
              'Phillips Exeter Academy\n'
              'Classics\n'
              'School year 2002\n'
              'Ardsley High School\n'
              'High School\n'
              'September 1998 - June 2000',
 'Favourite Quotes': '"Fortune favors the bold."\n'
                     '- Virgil, Aeneid X.284\n'
                     '\n'
                     '"All children are artists. The problem is how to remain '
                     'an artist once you grow up."\n'
                     '- Pablo Picasso\n'
                     '\n'
                     '"Make things as simple as possible but no simpler."\n'
                     '- Albert Einstein',
 'Name': 'Mark Zuckerberg',
 'Places lived': [{'link': '/profile.php?id=104022926303756&refid=17',
                   'text': 'Palo Alto, California',
                   'type': 'Current town/city'},
                  {'link': '/profile.php?id=105506396148790&refid=17',
                   'text': 'Dobbs Ferry, New York',
                   'type': 'Home town'}],
 'Work': 'Chan Zuckerberg Initiative\n'
         '1 December 2015 - Present\n'
         'Facebook\n'
         'Founder and CEO\n'
         '4 February 2004 - Present\n'
         'Palo Alto, California\n'
         'Bringing the world closer together.',
 'cover_photo': 'https://scontent.fakl1-2.fna.fbcdn.net/v/t31.18172-8/fr/cp0/e15/q65/19575079_10103832396388711_8894816584589808440_o.jpg?_nc_cat=109&ccb=1-3&_nc_sid=ed5ff1&_nc_ohc=JWr-Ynp5mrMAX-ZRcWx&_nc_ht=scontent.fakl1-2.fna&tp=14&oh=0a80f369d0ad77f855f91415b977f0fb&oe=60DED498',
 'id': '4',
 'profile_picture': 'https://scontent.fakl1-2.fna.fbcdn.net/v/t1.6435-1/fr/cp0/e15/q65/79515135_10111007623880301_5111576226921709568_n.jpg?_nc_cat=1&ccb=1-3&_nc_sid=dbb9e7&_nc_ohc=QAH7s-CmNV0AX9SYhfr&_nc_ht=scontent.fakl1-2.fna&tp=14&oh=e99db3b0a0b720fda3524ff26545bd35&oe=60E029D6'}
```
Note the `'id': '4'` field

**Reply by alvaroserrrano** [reply]
For my use case, the issue I have is that I know the name of the person, Mark Zuckerberg, and I would to get its username so that I can call get_profile(zuck).
I would like to get either zuck or its user_id from Mark Zuckerberg

**Reply by neon-ninja** [reply]
Names aren't unique. How is the scraper supposed to know which Mark you meant?

**Reply by alvaroserrrano** [reply]
More specifically, the name that I am referring to is the result of a call to get_posts from where I get 'username': post['username'], so the name is not actually the person's name but a username that contains some spaces. I guess username and account name are a different thing. The question to ask would be, how can I get an account's name from a username? I guess that usernames must be unique for Facebook

Sorry for the misconception

**Reply by neon-ninja** [reply]
`get_posts` provides you `user_id`s with the `user_id` field. You can use those. For example:

```python
post = next(get_posts("joebiden", pages=1, cookies="cookies.txt"))
print(post["user_id"], post["user_url"], post["username"])
```
returns
```python
7860876103 https://facebook.com/joebiden/?__tn__=C-R Joe Biden
```

**Reply by alvaroserrrano** [reply]
Thanks for it. However, for my use case I do not have the username ("joebiden"), so I cannot call the get_posts("joebiden"...) method. The only information that I know is the "Joe Biden", which, as shown on your example above, is the result of post["username"].
More specifically, my input is "Joe Biden" and I would like to get "joebiden" as the output so I can call get_profile("joebiden"). 
Also, do you recommend using cookies over credentials?
Thank you very much for your help. I appreciate it

**Reply by neon-ninja** [reply]
Again, how is the scraper supposed to know which Joe Biden you mean?
Look at all these: https://www.facebook.com/search/people/?q=joe%20biden
And all these: https://www.facebook.com/search/people/?q=mark%20zuckerberg

Either should be fine, I prefer cookies as then I'm not embedding passwords in plain text. Plus cookies results in 2 fewer requests which is marginally faster.
